lib/ansible/cli/vault.py

```diff
@@ -131,7 +131,7 @@ def post_process_args(self, options):
             raise AnsibleOptionsError("At most one input file may be used with the --output option")
 
         if options.action == 'encrypt_string':
-            if '-' in options.args or not options.args or options.encrypt_string_stdin_name:
+            if '-' in options.args or (not options.args and not options.encrypt_string_prompt) or options.encrypt_string_stdin_name:
```

Comment:
There should be a way to make these options mutually exclusive with the argparse lib. We should already have some options using that pattern.

Reply:
Hi! Absolutely, handy feature to understand - I'm using contributing to Ansible as a way to develop my own Python skills so that's another trick up my sleeve. :)

I'm not sure it's entirely possible to do away with the exclusivity check outside of argparse - ideally we'd need to retain the functionality that `-` also triggers a read from stdin. Unfortunately I couldn't get argparse to accept `-` as an argument without any additional arguments having to be added, despite setting the default action to be `store_true`.

I've pushed up a commit that moves the --prompt/--stdin-name arguments into an argparse mutual exclusivity group. The unit test needs updating to tolerate the new error in the meantime but just wanted to check I haven't missed anything as far as the issue with the stdin pseudo-argument goes.

If there's a cleaner way to go about this I'd love to hear about it

Cheers!

Reply:
you don't need argparse to handle `-` but  handle the case when the filename == `-`